File: src/main/java/com/kotoriSystems/entity/are_doko/LocationSubCategory.java
// package com.kotoriSystems.entity.are_doko;

// import javax.persistence.*;

// @Entity
// @Table(name = "location_sub_category")
// public class LocationSubCategory {

//     @Id
//     @GeneratedValue(strategy = GenerationType.IDENTITY)
//     private Long id;

//     @Column(nullable = false)
//     private String name;

//     @ManyToOne
//     @JoinColumn(name = "category_id", nullable = false)
//     private LocationCategory category;

//     public LocationSubCategory(Long id, String name, LocationCategory category) {
//       this.id = id;
//       this.name = name;
//       this.category = category;
//     }

//     public LocationSubCategory() {}

//     public LocationSubCategory(String name, LocationCategory category) {
//         this.name = name;
//         this.category = category;
//     }

//     public Long getId() {
//       return id;
//     }

//     public void setId(Long id) {
//       this.id = id;
//     }

//     public String getName() {
//       return name;
//     }

//     public void setName(String name) {
//       this.name = name;
//     }

//     public LocationCategory getCategory() {
//       return category;
//     }

//     public void setCategory(LocationCategory category) {
//       this.category = category;
//     }
// }

File: src/main/kotlin/com/kotorisystems/AreDoko/entity/LocationSubCategory.kt
package com.kotoriSystems.entity;

import javax.persistence.*;
// import com.kotoriSystems.entity.LocationCategory;

@Entity
@Table(name = "location_subcategory")
data class LocationSubCategory(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id : Long = 0,

    var name : String = "",

    // @ManyToOne(cascade = [CascadeType.PERSIST])
    // @JoinColumn(name = "category_id")
    // var category : LocationCategory? = null
)